src/imagecapturecore.cs

```diff
@@ -0,0 +1,901 @@
+		uint PtpObjectHandle { get; }
+	
+		[Export ("addedAfterContentCatalogCompleted")]
+		bool AddedAfterContentCatalogCompleted { [Bind ("wasAddedAfterContentCatalogCompleted")] get; }
+	}
+	
+	[BaseType (typeof(ICCameraItem))]
+	interface ICCameraFolder {
+
+		[Export ("contents")]
+		ICCameraItem[] Contents { get; }
+	}
+	
+	[BaseType (typeof(ICCameraItem))]
+	interface ICCameraFile {
+
+		[Export ("fileSize")]
+		long FileSize { get; }
+	
+		[Export ("orientation", ArgumentSemantic.Assign)]
+		ICExifOrientationType Orientation { get; set; }
+	
+		[Export ("duration")]
+		double Duration { get; }
+	
+
+		[Export ("sidecarFiles")]
+		ICCameraItem[] SidecarFiles { get; }
+	}
+
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))] 
+	interface ICCameraDeviceDelegate : ICDeviceDelegate {
+
+		[Abstract]
+		[Export ("cameraDevice:didAddItem:")]
+		void DidAddItem (ICCameraDevice camera, ICCameraItem item);
+	
+		[Abstract]
+		[Export ("cameraDevice:didRemoveItem:")]
+		void DidRemoveItem (ICCameraDevice camera, ICCameraItem item);
+	
+		[Abstract]
+		[Export ("cameraDevice:didRenameItems:")]
+		void DidRenameItems (ICCameraDevice camera, ICCameraItem[] items);
+	
+		[Abstract]
+		[Export ("cameraDevice:didCompleteDeleteFilesWithError:")]
+		void DidCompleteDeleteFiles (ICCameraDevice scanner, [NullAllowed] NSError error);
+	
+		[Abstract]
+		[Export ("cameraDeviceDidChangeCapability:")]
+		void DidChangeCapability (ICCameraDevice camera);
+	
+		[Abstract]
+		[Export ("cameraDevice:didReceiveThumbnailForItem:")]
+		void DidReceiveThumbnail (ICCameraDevice camera, ICCameraItem forItem);
+	
+		[Abstract]
+		[Export ("cameraDevice:didReceiveMetadataForItem:")]
+		void DidReceiveMetadata (ICCameraDevice camera, ICCameraItem forItem);
+	
+		[Abstract]
+		[Export ("cameraDevice:didReceivePTPEvent:")]
+		void DidReceivePtpEvent (ICCameraDevice camera, NSData eventData);
+	
+		[Abstract]
+		[Export ("deviceDidBecomeReadyWithCompleteContentCatalog:")]
+		void DidBecomeReadyWithCompleteContentCatalog (ICDevice device);
+	
+		[Export ("cameraDevice:didAddItems:")]
+		void DidAddItems (ICCameraDevice camera, ICCameraItem[] items);
+	
+		[Export ("cameraDevice:didRemoveItems:")]
+		void DidRemoveItems (ICCameraDevice camera, ICCameraItem[] items);
+	
+		[Export ("cameraDevice:shouldGetThumbnailOfItem:")]
+		bool ShouldGetThumbnail (ICCameraDevice cameraDevice, ICCameraItem ofItem);
+	
+		[Export ("cameraDevice:shouldGetMetadataOfItem:")]
+		bool ShouldGetMetadata (ICCameraDevice cameraDevice, ICCameraItem ofItem);
+	}
+
+	interface IICCameraDeviceDownloadDelegate {}
+
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof(NSObject))]
+	interface ICCameraDeviceDownloadDelegate {
+
+		[Export ("didDownloadFile:error:options:contextInfo:")]
+		void DidDownloadFile (ICCameraFile file, [NullAllowed] NSError error, [NullAllowed] NSDictionary<NSString, NSObject> options, [NullAllowed] IntPtr contextInfo);
```

Comment:
Do we know the possible values of the keys for the options dict? In that case we could use a strong dict. If not, as with the other comment, we should use NSDictionary to be able to use a strong dict whenever we do know them.

Reply:
We can't do strong dictionaries in protocol members, so having the generic dict here is the best we can do right now